template_handler/cdk.py

```diff
@@ -0,0 +1,308 @@
+
+    def _cmd_exists(self, cmd: str) -> bool:
+        """
+        Checks whether a specified CLI command exists.
+
+        Args:
+            cmd: str - The name of the CLI command to check
+
+        Returns:
+            bool
+        """
+
+        cmd_exists = shutil.which(cmd) is not None
+        self.logger.debug(f"command '{cmd}' exists: {cmd_exists}")
+        return cmd_exists
+
+    def _node_package_exists(self, package: str) -> bool:
+        """
+        Checks whether a specifie Node package exists either in the workspace or global scope.
+
+        Args:
+            package: str - The name of the node package to check
+
+        Returns:
+            bool
+        """
+
+        package_exists = False
+        workspace_result = subprocess.run(
+            f'npm list {package}',
+            shell=True,
+            capture_output=True
+        )
+        self.logger.debug(f"Workspace NPM package '{package}' exists: {not bool(workspace_result.returncode)}")
+
+        if workspace_result.returncode == 0:
+            package_exists = True
+        else:
+            global_result = subprocess.run(
+                f'npm --global list {package}',
+                shell=True,
+                capture_output=True
+            )
+            self.logger.debug(f"f'Global NPM package '{package}' exists: {not bool(global_result.returncode)}")
+
+            if global_result.returncode == 0:
+                package_exists = True
+
+        return package_exists
+
+    def _check_prerequisites(self) -> None:
+        """
+        Checks the command and Node package requirements for the handler.
+
+        Raises:
+            SceptreException: Command prerequisite not found
+            SceptreException: Node Package prerequisite not found
+        """
+
+        # Check Command Prerequisites
+        cmd_prerequisites = [
+            'node',
+            'npx'
+        ]
+        for cmd_prerequisite in cmd_prerequisites:
+            if not self._cmd_exists(cmd_prerequisite):
+                raise exceptions.SceptreException(f"{self.name} - Command prerequisite '{cmd_prerequisite}' not found")
+
+        # Check Node Package Prerequisites
+        node_prerequisites = [
+            'cdk-assets'
+        ]
+        for node_prerequisite in node_prerequisites:
+            if not self._node_package_exists(node_prerequisite):
+                raise exceptions.SceptreException(f"{self.name} - Node Package prerequisite '{node_prerequisite}' not found")
+
+    def _import_python_template_module(self, cdk_template_path: str) -> ModuleType:
+        """
+        Import the CDK Python template module.
+
+        Args:
+            cdk_template_path: str - The path of the CDK Template
+
+        Returns:
+            A ModuleType object containing the imported CDK Python module
+
+        Raises:
+            SceptreException: Template File not found
+            SceptreException: importlib general exception
+        """
```

Comment:
I just want to say I appreciate your use of Google-style docstrings. They're my preference as well. Sphinx is ok... but ever since py3 introduced type annotations, the sphinx format isn't really that helpful.